torchvtk/augmentation/DictTransform.py

```diff
@@ -0,0 +1,150 @@
+import math
+import random
+from abc import abstractmethod
+
+import numpy as np
+import torch
+from torch.nn import functional as f
+
+
+class DictTransform(object):
+
+    def __init__(self, device="cpu", apply_on=["vol", "mask"], dtype=torch.float32, batch_transform=False):
+        super().__init__()
+
+        self.device = device
+        self.apply_on = apply_on
+        self.dtype = dtype
+        self.batch_transform = batch_transform
+
+    @abstractmethod
+    def transform(self, data): pass
+
+    def __call__(self, data):
+
+        for key in self.apply_on:
+            tmp = data[key]
+
+            if not torch.is_tensor(tmp):
+                tmp = torch.from_numpy(tmp)
+
+            # to GPU.
+            if self.device == "cuda":
+                tmp = tmp.to(self.device)
+
+            if self.dtype is torch.float16:
+                tmp = tmp.to(torch.float32)
+
+            if self.batch_transform is False:
+                tmp = self.transform(tmp)
+            else:
+                batch = []
+                for x in range(tmp.shape[0]):
+                    # get batch thing
+                    tmptensor = tmp[x, ...]
+                    # transform
+                    batch.append(self.transform(tmptensor))
+                # stack tensor back
+                tmp = torch.stack(batch, dim=0)
+
+            if self.dtype is torch.float16:
+                tmp = tmp.to(torch.float16)
+            elif self.dtype is not torch.float16 or torch.float32:
+                tmp = tmp.to(torch.float32)
+            tmp = tmp.to(self.device)
+            # back on device?
+            data[key] = tmp
+        return data
+
+
+class NoiseDictTransform(DictTransform):
+
+    def __init__(self, device, noise_variance=(0.001, 0.05), apply_on=["vol"], dtype=torch.float32, batch_transform=False):
+
+        self.device = device
+        self.noise_variance = noise_variance
+        DictTransform.__init__(self, self.device, apply_on=apply_on, dtype=dtype, batch_transform=batch_transform)
+
+    def transform(self, data):
+        #todo allow for batch
+        variance = random.uniform(self.noise_variance[0], self.noise_variance[1])
+        noise = np.random.normal(0.0, variance, size=data.shape)
+        noise = torch.from_numpy(noise)
+        noise = noise.to(self.device)
+        sample = torch.add(data, noise)
+        return sample
+
+
+class BlurDictTransform(DictTransform):
+
+    def __init__(self, apply_on=["vol"], dtype=torch.float32, device= "cpu", channels=1, kernel_size=(3, 3, 3), sigma=1, batch_transform=False):
```

Comment:
best use `**kwargs` for all params that go to DictTransform.
we'll have to find out how to still get proper documentation for the parameters with the kwargs